bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -0,0 +1,258 @@
+    private static final DateTimeFormatter ENEDIS_DATE_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static final int HTTP_DEFAULT_TIMEOUT_MS = (int) TimeUnit.SECONDS.toMillis(5);
+    // private static final Builder LOGIN_BODY_BUILDER = new FormBody.Builder().add("encoded", "true")
+    // .add("gx_charset", "UTF-8").add("SunQueryParamsString",
+    // Base64.getEncoder().encodeToString("realm=particuliers".getBytes(StandardCharsets.UTF_8)));
+
+    // private final OkHttpClient client = new OkHttpClient.Builder().followRedirects(false)
+    // .cookieJar(new LinkyCookieJar()).build();
+    private final Gson GSON = new Gson();
+
+    private @NonNullByDefault({}) ScheduledFuture<?> refreshJob;
+    private final WeekFields weekFields;
+
+    public LinkyHandler(Thing thing, LocaleProvider localeProvider) {
+        super(thing);
+        weekFields = WeekFields.of(localeProvider.getLocale());
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Linky handler.");
+
+        LinkyConfiguration config = getConfigAs(LinkyConfiguration.class);
+        logger.debug("config username = '{}'", config.username);
+
+        try {
+            // Request requestLogin = new Request.Builder().url(LOGIN_BASE_URI)
+            // .post(LOGIN_BODY_BUILDER.add("IDToken1", config.username).add("IDToken2", config.password).build())
+            // .build();
+            // client.newCall(requestLogin).execute().close();
+            String requestContent = String.format(LOGIN_BODY_BUILDER,
+                    Base64.getEncoder().encodeToString("realm=particuliers".getBytes(StandardCharsets.UTF_8)),
+                    config.username, config.password);
+            InputStream stream = new ByteArrayInputStream(requestContent.getBytes(StandardCharsets.UTF_8));
+            logger.debug("executeUrl POST {} requestContent {}", LOGIN_BASE_URI, requestContent);
+            HttpUtil.executeUrl("POST", LOGIN_BASE_URI, stream, "application/x-www-form-urlencoded",
+                    HTTP_DEFAULT_TIMEOUT_MS);
+            stream.close();
+
+            // first call to test connection and to get needed cookies
+            // getEnedisInfo(DAILY, LocalDate.now(), LocalDate.now());
+
+            updateStatus(ThingStatus.ONLINE);
+            refreshJob = scheduler.scheduleWithFixedDelay(() -> {
+                updateLinkyData();
+            }, 0, config.refreshInterval, TimeUnit.MINUTES);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * Request new data and updates channels
+     */
+    private void updateLinkyData() {
+        final LocalDate today = LocalDate.now();
+
+        LocalDate rangeStart = today.minusDays(9);
+        EnedisInfo result = getEnedisInfo(DAILY, rangeStart, today);
```

Comment:
Why requesting data for only 10 days while you want to compute last week and current week ?
I have the feeling you could miss some data for the last week depending on the current week day.

Reply:
You're right. This is a leftover of my initial tests. I've to review this algorithm.